lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -825,10 +859,12 @@ class MongoClientInterface {
                             { error: err.message });
                         return next(errors.InternalError);
                     }
-                    if (!doc) {
+                    if (!doc && params && params.versionId) {
```

Comment:
I don't think we should change the behavior here (I may be wrong):

When we get an object without a version ID (so implicitly the latest version), whether it's a non-existing object or a delete marker, it should always return `NoSuchKey` I believe, per the API. So everything we have to do is check the status of the master key, even in v1.

Reply:
I do agree with the fact that with or without the modification at the end of the day we'll get the same response from Cloudserver (A NoSuchKey error), but i didn't want to change the behaviour of the function cause before, if the master was a delete marker it still got returned and then the logic is handled in Cloudserver.